:sparkles: Add authentication

UZ9 opened the pull request:
## PR Title
Issue Number(s): #number_of_issue

What does this PR change and why?
### Checklist
- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes
- Database change / migration to run
- Environment config change
- Breaking API change

### Related PRs
PR Number(s): #number_of_pr

### How To Test
The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

UZ9 committed:
:sparkles: Add authentication

afazio1 reviewed (approved):
lgtm. love this new feat

afazio1 commented:
Still need to figure out dynamic NEXTAUTH_URL for deploy previews

afazio1 committed:
Merge branch 'ryder-add-auth-back' of https://github.com/GTBitsOfGood/canine-assistants into ryder-add-auth-back

afazio1 committed:
add netlify.toml file

UZ9 committed:
:bug: Fix SessionProvider not receiving valid session

UZ9 committed:
:construction: Add temporary debug for NEXTAUTH_URL

UZ9 committed:
:construction: Temporarily remove newUser and error page redirects

UZ9 committed:
:construction: Remove netlify.toml

afazio1 committed:
downgrade nextjs netlify plugin to 4.36.1

UZ9 committed:
:construction: Mirror [...nextauth] in middleware

UZ9 committed:
Merge branch 'ryder-add-auth-back' of https://github.com/GTBitsOfGood/canine-assistants into ryder-add-auth-back

afazio1 committed:
try custom redirect callback

UZ9 committed:
:sparkles: Revamp Login + Credentials Auth

- Added bcrypt
- Added method for verifying whether a user has the right credentials
- Removed old console statements
- Added passwordHashed to User schema
- Reverted middleware.js back to original state
- Added CredentialsProvider using JWT to [...nextauth].js
- Revamped login page to show new login method
- Fixed old art inconsistencies between the login and figma pages

afazio1 committed:
set google redirect false

afazio1 committed:
remove custom nextjs-plugin version

UZ9 committed:
:bug: Fix multiple redirecting bugs

UZ9 committed:
:construction: Add debug marker to solve credentials issue

afazio1 committed:
Merge branch 'dev' into ryder-add-auth-back

afazio1 committed:
fix seeder
